btx/processing/rawimagetimetool.py

```diff
@@ -0,0 +1,575 @@
+        """
+        ## @var expmt
+        # (str) Experiment accession name
+        self.expmt = expmt
+
+        ## @var hutch
+        # (str) Experimental hutch. Extracted from expmt. Needed for camera/stage accession.
+        self.hutch = self.expmt[:3]
+
+        ## @var _model
+        # (list) List of calibration model polynomial coefficients. Empty if not calibrated.
+        self._model = []
+
+        ## @var savedir
+        # (str) Directory to save output files to. Figures may be placed in a subdirectory of this one.
+        self.savedir = savedir
+
+    def open_run(self, run: str):
+        """! Open the psana DataSource and time tool detector for the specified
+        run. This method MUST be called before calibration, analysis etc. The
+        psana DataSource and Detector objects are stored in instance variables
+
+        @param run (str) Run number (e.g. 5) or range of runs (e.g. 6-19).
+        """
+
+        ## @var ds
+        # (psana.DataSource) DataSource object for accessing data for specified runs.
+        self.ds = psana.MPIDataSource(f'exp={self.expmt}:run={run}')
+
+        # Detector lookup needs to be modified for each hutch!
+        # Considering only MFX right now
+        ## @var det
+        # (psana.Detector) Detector object corresponding to the time tool camera.
+        # self.det = psana.Detector(self.get_detector_name(run), self.ds.env())
+        self.det = psana.Detector(self.get_tt_name(), self.ds.env())
+
+    def get_tt_name(self) -> str:
+        """! Run psana.MPIDataSource.detnames() to get associated detectors,
+        process the output and find the name of the timetool detector.
+
+        @return detname (str) Name of the timetool detector (full or alias).
+        """
+        for tup in self.ds.detnames():
+            for name in tup:
+                if any(alias in name.lower() for alias in ['opal', 'timetool']):
+                    return name
+
+    def get_detector_name(self, run:str) -> str:
+        """! This method is deprecated.
+        Run detnames from shell and determine the time tool detector name.
+
+        @param run (str) Experiment run to check detectors for
+        @return detname (str) Name of the detector (full or alias)
+        """
+        # Define command to run
+        prog = 'detnames'
+        arg = f'exp={self.expmt}:run={run}'
+        cmd = [prog, arg]
+
+        # Run detnames, save output
+        output = subprocess.check_output(cmd)
+
+        # Convert to list and strip some characters
+        output = str(output).split()
+        output = [o.strip('\\n') for o in output]
+
+        # Select detectors related to timetool (Opal, e.g.)
+        # Need to see if this actually works for all hutches, seems to be good
+        # for cxi and mfx
+        ttdets = [det for det in output if "opal" in det.lower() or "timetool" in det.lower()]
+
+        # Return the first entry (could use any)
+        return ttdets[0]
```

Comment:
Let's not keep that one since it is deprecated.

However, let's mention how to use `detnames` in the documentation, and maybe illustrate with this example?